### ISSUE TITLE
[jss-template] Using template literals in JSS yields "Malformed CSS string" warning

### ISSUE THREAD

**Issue by cezarderevlean** [cssinjs/jss#663]
Because it seems to be related to JSS, I'm adding below the following issue: https://github.com/mui-org/material-ui/issues/9932

---

I added `jss-preset-default` in order to be able to use template literals in JSS, among other stuff.

```jsx
import { create } from 'jss';
import jssPreset from 'jss-preset-default';
import JssProvider from 'react-jss/lib/JssProvider';

const jss = create(jssPreset());

...

<JssProvider jss={jss}>
    <App />
</JssProvider>
```
```jsx
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
    test: `
        color: red;
        background: blue;
    `
});

class App extends Component {
    render() {
        return (
            <div className={this.props.classes.test}>
                Lorem ipsum dolor
            </div>
        );
    }
}

export default withStyles(styles)(App);
```

## Expected Behavior
It should make the text in the div red and the div background blue, without any console errors.

## Current Behavior
The styles apply correctly and work fine, but this error shows up in console (in red):
`Warning: Malformed CSS string "        "`

| warning | @ | browser.js?1af0:49 |
| -- | -- | -- |
| exports.default | @ | parse.js?fbfc:31 |
| onProcessRule | @ | index.js?ff0a:15 |
| onProcessRule | @ | PluginsRegistry.js?e077:56 |
| process | @ | RuleList.js?5e19:147 |
| GlobalContainerRule | @ | index.js?9c36:36 |
| onCreateRule | @ | index.js?9c36:162 |
| onCreateRule | @ | PluginsRegistry.js?e077:39 |
| createRule | @ | createRule.js?c07d:33 |
| add | @ | RuleList.js?5e19:88 |
| StyleSheet | @ | StyleSheet.js?6b95:40 |
| createStyleSheet | @ | Jss.js?370d:123 |
| attach | @ | withStyles.js?deb5:282 |
| attach | @ | createPrototypeProxy.js?6ea8:44 |
| componentWillReceiveProps | @ | withStyles.js?deb5:238 |
| componentWillReceiveProps | @ | createPrototypeProxy.js?6ea8:44 |
| callComponentWillReceiveProps | @ | react-dom.development.js?cada:6389 |
| updateClassInstance | @ | react-dom.development.js?cada:6575 |
| updateClassComponent | @ | react-dom.development.js?cada:7848 |
| beginWork | @ | react-dom.development.js?cada:8225 |
| performUnitOfWork | @ | react-dom.development.js?cada:10224 |
| workLoop | @ | react-dom.development.js?cada:10288 |
| callCallback | @ | react-dom.development.js?cada:542 |
| invokeGuardedCallbackDev | @ | react-dom.development.js?cada:581 |
| invokeGuardedCallback | @ | react-dom.development.js?cada:438 |
| renderRoot | @ | react-dom.development.js?cada:10366 |
| performWorkOnRoot | @ | react-dom.development.js?cada:11014 |
| performWork | @ | react-dom.development.js?cada:10967 |
| requestWork | @ | react-dom.development.js?cada:10878 |
| scheduleWorkImpl | @ | react-dom.development.js?cada:10732 |
| scheduleWork | @ | react-dom.development.js?cada:10689 |
| scheduleTopLevelUpdate | @ | react-dom.development.js?cada:11193 |
| updateContainer | @ | react-dom.development.js?cada:11231 |
| renderSubtreeIntoContainer | @ | react-dom.development.js?cada:15229 |
| render | @ | react-dom.development.js?cada:15290 |
| render | @ | index.jsx?a980:36 |
| (anonymous) | @ | index.jsx?a980:55 |
| (anonymous) | @ | index.jsx?a980:55 |
| hotApply | @ | main.js:582 |
| (anonymous) | @ | main.js:290 |
| Promise resolved (async) |   |   |
| hotUpdateDownloaded | @ | main.js:289 |
| hotAddUpdateChunk | @ | main.js:266 |
| webpackHotUpdateCallback | @ | main.js:8 |
| (anonymous) | @ | 0.7715afa….hot-update.js:1 |

## Steps to Reproduce (for bugs)
https://codesandbox.io/s/jpm8kx6245

1. Use code above
2. Check console

## Context
I'm trying to use template literals in JSS (with the help of `jss-template` which is included in `jss-preset-default`).

## Environment

| Tech         | Version |
|--------------|---------|
├─ jss-preset-default | 4.0.1
├─ jss | 9.5.1
├─ material-ui | 1.0.0-beta.29
├─ react-jss | 8.2.1
└─ react | 16.2.0
Browser: Chrome | 63.0.3239.132

**Comment by cezarderevlean** [comment]
Fixed in v1.0.1

**Comment by cezarderevlean** [comment]
Apparently the issue still persists.

This does *not* yield warning:
```
const styles = theme => ({
    '@global': {
        body: `
            overflow: hidden !important;
            padding-right: 0 !important;
            color: ${colors.grey[800]};
            background: ${colors.grey[50]};
            font-family: Roboto;
`,
        '#root': `
            overflow-y: scroll;
            height: 100vh;
`,
        'a, a:hover, a:visited, a:active, a:focus': `
            text-decoration: none;
`,
    }
});
```

This *does* yield 3 warnings:
```
const styles = theme => ({
    '@global': {
        body: `
            overflow: hidden !important;
            padding-right: 0 !important;
            color: ${colors.grey[800]};
            background: ${colors.grey[50]};
            font-family: Roboto;
        `,
        '#root': `
            overflow-y: scroll;
            height: 100vh;
        `,
        'a, a:hover, a:visited, a:active, a:focus': `
            text-decoration: none;
        `,
    }
});
```
So it seems there's something with that indenting whitespace before the closing backtick for each selector.

**Comment by kof** [comment]
Interesting, lest see.

**Comment by kof** [comment]
works for me https://codesandbox.io/s/ovkrp7lkxz

**Comment by cezarderevlean** [comment]
I'm using `jss-preset-default`. Does it require the updated `jss-template` dependency?
I've just tested in codesandbox with `jss-preset-default` and it does show the warnings.

**Comment by kof** [comment]
Oh, I forgot jss-preset-default also has a package-lock

**Comment by kof** [comment]
hmm, mb preset shouldn't use a package-lock

**Comment by kof** [comment]
for the sake of not updating it each time

**Comment by kof** [comment]
released new jss-preset-default v4.1.0
